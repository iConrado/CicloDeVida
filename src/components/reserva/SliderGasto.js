import React from 'react';

import { 
  View, 
  Text } from 'react-native';
import Slider from 'react-native-slider';

import styles from '../functions/styles';
import Ciclo from '../functions/Ciclo';
import monetizar from '../functions/monetizar';

const C = new Ciclo();

export default class SliderGasto extends React.Component {
  constructor(props) {
    super(props);
    this.state = { tmpGasto: props.inicial ? props.inicial : 0 }; 
  }

  maximumSlider() {
    const max = parseInt(C.getSalLiq(), 10);
    return max;
  }

  render() {
    return (
      <View style={styles.viewVertical}>
        <View style={styles.viewHorizontal}>
          <Text style={styles.label}>Informe seu gasto mensal: </Text>
        </View>
        <View style={styles.viewHorizontal}>
          <Slider
            style={styles.reserva_slider}
            minimumValue={0}
            maximumValue={this.maximumSlider()}
            step={100}
            minimumTrackTintColor='#14567A'
            thumbTintColor='#14567A'
            value={this.state.tmpGasto}
            onValueChange={(value) => this.setState({ tmpGasto: value })}
            onSlidingComplete={(value) => this.props.retorno(value)}
          />
          <View style={styles.reserva_viewCentral}>
            <Text style={styles.reserva_txDireita}>{monetizar(this.state.tmpGasto)}</Text>
          </View>
        </View>
      </View>
    );
  }
}
