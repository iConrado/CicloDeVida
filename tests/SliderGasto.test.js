import 'react-native';
import renderer from 'react-test-renderer';

import React from 'react';
import SliderGasto from '../src/components/reserva/SliderGasto';

it('renders correctly', () => {
  const rendered = renderer.create(<SliderGasto />).toJSON();
  expect(rendered).toMatchSnapshot();
});
