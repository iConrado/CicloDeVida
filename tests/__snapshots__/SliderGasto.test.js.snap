// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderGasto renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderGasto
    inicial={50}
    retorno={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Informe seu gasto mensal: 
          </Text>
        </View>,
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Slider
            animationType="timing"
            debugTouchArea={false}
            maximumTrackTintColor="#b3b3b3"
            maximumValue={0}
            minimumTrackTintColor="#14567A"
            minimumValue={0}
            onSlidingComplete={[Function]}
            onValueChange={[Function]}
            step={100}
            style={
              Object {
                "height": 40,
                "width": "70%",
              }
            }
            thumbTintColor="#14567A"
            thumbTouchSize={
              Object {
                "height": 40,
                "width": 40,
              }
            }
            value={50}
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              R$ 50
            </Text>
          </View>
        </View>,
      ],
      "style": Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Informe seu gasto mensal: 
          </Text>,
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Informe seu gasto mensal: ",
            "style": Object {
              "color": "#666",
            },
          },
          "ref": null,
          "rendered": "Informe seu gasto mensal: ",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={0}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={100}
              style={
                Object {
                  "height": 40,
                  "width": "70%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />,
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                R$ 50
              </Text>
            </View>,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationType": "timing",
              "debugTouchArea": false,
              "maximumTrackTintColor": "#b3b3b3",
              "maximumValue": 0,
              "minimumTrackTintColor": "#14567A",
              "minimumValue": 0,
              "onSlidingComplete": [Function],
              "onValueChange": [Function],
              "step": 100,
              "style": Object {
                "height": 40,
                "width": "70%",
              },
              "thumbTintColor": "#14567A",
              "thumbTouchSize": Object {
                "height": 40,
                "width": 40,
              },
              "value": 50,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                R$ 50
              </Text>,
              "style": Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "R$ 50",
                "style": Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              },
              "ref": null,
              "rendered": "R$ 50",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Informe seu gasto mensal: 
            </Text>
          </View>,
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={0}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={100}
              style={
                Object {
                  "height": 40,
                  "width": "70%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  }
                }
              >
                R$ 50
              </Text>
            </View>
          </View>,
        ],
        "style": Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              style={
                Object {
                  "color": "#666",
                }
              }
            >
              Informe seu gasto mensal: 
            </Text>,
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Informe seu gasto mensal: ",
              "style": Object {
                "color": "#666",
              },
            },
            "ref": null,
            "rendered": "Informe seu gasto mensal: ",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={0}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={100}
                style={
                  Object {
                    "height": 40,
                    "width": "70%",
                  }
                }
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />,
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "width": "30%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  R$ 50
                </Text>
              </View>,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animationType": "timing",
                "debugTouchArea": false,
                "maximumTrackTintColor": "#b3b3b3",
                "maximumValue": 0,
                "minimumTrackTintColor": "#14567A",
                "minimumValue": 0,
                "onSlidingComplete": [Function],
                "onValueChange": [Function],
                "step": 100,
                "style": Object {
                  "height": 40,
                  "width": "70%",
                },
                "thumbTintColor": "#14567A",
                "thumbTouchSize": Object {
                  "height": 40,
                  "width": 40,
                },
                "value": 50,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  style={
                    Object {
                      "alignSelf": "center",
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "100%",
                    }
                  }
                >
                  R$ 50
                </Text>,
                "style": Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "width": "30%",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "R$ 50",
                  "style": Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "100%",
                  },
                },
                "ref": null,
                "rendered": "R$ 50",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
